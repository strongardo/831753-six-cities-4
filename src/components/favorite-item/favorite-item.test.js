import React from "react";
import renderer from "react-test-renderer";
import FavoriteItem from "./favorite-item.jsx";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {Router} from 'react-router-dom';
import history from '../../history.js';

const mockStore = configureStore([]);

const city =
  {
    name: `Paris`,
    id: `Paris`,
  };

const offers = [
  {
    name: `The house among olive`,
    descriptions: `The house among olive`,
    type: `house`,
    owner: {
      url: `/img/avatar-angelina.jpg`,
      name: `Angelina`,
      isSuper: false,
    },
    advantages: [`Laptop friendly workspace`],
    price: 813,
    url: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`,
    urls: [`https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`],
    starsCount: 4.2,
    bedroomsCount: 3,
    guestsCount: 6,
    isPremium: true,
    isFavorite: true,
    coordinates: [52.3909553943508, 4.85309666406198],
    city: {
      name: `Paris`,
      coordinates: [52.38333, 4.9],
      zoom: 12,
      id: `Paris`,
    },
    id: 1,
  }
];

it(`Should FavoriteItem render correctly`, () => {
  const store = mockStore({});
  const tree = renderer
    .create(
        <Provider store={store}>
          <Router history={history}>
            <FavoriteItem
              city={city}
              offers={offers}
            />
          </Router>
        </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
