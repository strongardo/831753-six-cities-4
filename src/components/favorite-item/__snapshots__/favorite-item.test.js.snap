// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FavoriteItem render correctly 1`] = `
<li
  className="favorites__locations-items"
>
  <div
    className="favorites__locations locations locations--current"
  >
    <div
      className="locations__item"
    >
      <a
        className="locations__item-link"
        href="#"
      >
        <span>
          Paris
        </span>
      </a>
    </div>
  </div>
  <div
    className="favorites__places"
  >
    <article
      className="favorites__card place-card"
    >
      <div
        className="favorites__image-wrapper place-card__image-wrapper"
      >
        <a
          className="place-card__link"
          href="/offer/1"
          onClick={[Function]}
        >
          <img
            alt="Place image"
            className="place-card__image"
            height={110}
            src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg"
            width={150}
          />
        </a>
      </div>
      <div
        className="favorites__card-info place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              813
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button button place-card__bookmark-button--active"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height={19}
              width={18}
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              In bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "84%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            className="place-card__link"
            href="/offer/1"
            onClick={[Function]}
          >
            The house among olive
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          house
        </p>
      </div>
    </article>
  </div>
</li>
`;
